ir/dump.h

```diff
@@ -69,6 +69,15 @@ inline std::ostream &operator<<(std::ostream &out, const Dump &d) {
     return out;
 }
 
+struct DumpPipe : public Inspector {
+    const char *heading;
+    DumpPipe() : heading(nullptr) {}
+    explicit DumpPipe(const char *h) : heading(h) {}
+#if ENABLE_DUMP_PIPE
+    bool preorder(const IR::Node *pipe) override;
```

Comment:
if `ENABLE_DUMP_PIPE` is false, should have this preorder method run and just return false, so as to not needlessly visit the entire IR tree and do nothing.  So remove this `#if` ...

Reply:
Good point. Fixed.

Reply:
I don't see a push with this change?